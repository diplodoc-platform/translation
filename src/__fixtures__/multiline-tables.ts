const markdown = `\
#|
||**Первый Заголовок**|**Второй Заголовок**||
||текст первой колонки|текст второй колонки||
|#

- текст перед таблицей

  #|
  ||
  Первый Заголовок
  |
  Второй Заголовок
  ||
  ||
  текст первой колонки
  |
  текст второй колонки
  ||
  |#

  текст после таблицы
`;

const skeleton = `\
#|
||**%%%1%%%**|**%%%2%%%**||
||%%%3%%%|%%%4%%%||
|#

- %%%5%%%

  #|
  ||
  %%%6%%%
  |
  %%%7%%%
  ||
  ||
  %%%8%%%
  |
  %%%9%%%
  ||
  |#

  %%%10%%%
`;

const translations = new Map<string, string>([
    ['1', 'First Heading'],
    ['2', 'Second Heading'],
    ['3', 'first cell text'],
    ['4', 'second cell text'],
    ['5', 'text before table'],
    ['6', 'First Heading'],
    ['7', 'Second Heading'],
    ['8', 'first cell text'],
    ['9', 'second cell text'],
    ['10', 'text after table'],
]);

export {markdown, skeleton, translations};
export default {markdown, skeleton, translations};
