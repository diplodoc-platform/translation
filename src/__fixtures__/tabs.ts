const markdown = `\
{% list tabs %}

- Первый Таб

  Текст первого таба

  - список внутри таба
        
    контент внутри списка
- Второй Таб
  
  Текст второго таба

{% endlist %}

{% list tabs %}

* Первый Таб во второй таб секции

  контент первого таба во второй таб секции

{% endlist %}
`;

const skeleton = `\
{% list tabs %}

- %%%1%%%

  %%%2%%%

  - %%%3%%%
        
    %%%4%%%
- %%%5%%%
  
  %%%6%%%

{% endlist %}

{% list tabs %}

* %%%7%%%

  %%%8%%%

{% endlist %}
`;

const translations = new Map<string, string>([
    ['1', 'First tab'],
    ['2', 'text of the first tab'],
    ['3', 'list inside the tab'],
    ['4', 'content inside the list'],
    ['5', 'Second tab'],
    ['6', 'text of the second tab'],
    ['7', 'First tab inside the second tab section'],
    ['8', 'content of the first tab inside the second tab section'],
]);

export {markdown, skeleton, translations};
export default {markdown, skeleton, translations};
