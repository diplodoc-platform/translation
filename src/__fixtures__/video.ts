const markdown = `\
@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) текст после
@[youtube](dQw4w9WgXcQ)
`;

const skeleton = `\
@[youtube](https://www.youtube.com/video?v=dQw4w9WgXcQ) %%%1%%%
@[youtube](dQw4w9WgXcQ)
`;

const translations = new Map<string, string>([['1', 'text after']]);

export {markdown, skeleton, translations};
export default {markdown, skeleton, translations};
