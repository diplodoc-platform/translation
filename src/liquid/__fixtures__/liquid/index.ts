export * from './fixture';
export * from './text';
export * from './condition';
export * from './functions';
export * from './filter';
export * from './variable';
export * from './loop';
export * as default from './index';
