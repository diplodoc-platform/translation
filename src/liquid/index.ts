export {Tokenizer} from './tokenizer';
export * from './token';
export * from './renderer';
export * from './index';
