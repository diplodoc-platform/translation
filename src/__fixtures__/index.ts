import basic from 'src/__fixtures__/basic';
import variables from 'src/__fixtures__/variables';
import filters from 'src/__fixtures__/filters';
import functions from 'src/__fixtures__/functions';
import conditionals from 'src/__fixtures__/conditionals';
import loops from 'src/__fixtures__/loops';
import meta from 'src/__fixtures__/meta';
import notes from 'src/__fixtures__/notes';
import cuts from 'src/__fixtures__/cuts';
import gfmTables from 'src/__fixtures__/gfm-tables';
import checkbox from 'src/__fixtures__/checkbox';
import anchors from 'src/__fixtures__/anchors';
import imsize from 'src/__fixtures__/imsize';
import links from 'src/__fixtures__/links';
import includes from 'src/__fixtures__/includes';
import tabs from 'src/__fixtures__/tabs';
import video from 'src/__fixtures__/video';

export {
    basic,
    variables,
    filters,
    functions,
    conditionals,
    loops,
    meta,
    notes,
    cuts,
    gfmTables,
    checkbox,
    anchors,
    imsize,
    links,
    includes,
    tabs,
    video,
};
export default {
    basic,
    variables,
    filters,
    functions,
    conditionals,
    loops,
    meta,
    notes,
    cuts,
    gfmTables,
    checkbox,
    anchors,
    imsize,
    links,
    includes,
    tabs,
    video,
};
