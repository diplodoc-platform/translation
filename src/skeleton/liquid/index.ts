import {TokenSubType, TokenType} from './token';
import {token} from 'src/utils';
import {vr, vrre} from 'src/symbols';

export type Configuration = {
  specification?: Specification;
};

export type Specification = Array<SpecificationEntry>;
export type SpecificationEntry = [RegExp, TokenType, TokenSubType] | [RegExp, TokenType];

const specification_: Specification = [
  // Conditions
  // If statement
  [/^\{%\s*if[^%}]+?\s*%}/, 'liquid', 'Literal'],
  // Else statement
  [/^\{%\s*else\s*%\}/, 'liquid', 'Literal'],
  // EndIf statement
  [/^\{%\s*endif\s*%\}/, 'liquid', 'Literal'],
  // Changelog
  [/^\{%\s*changelog\s*%\}/, 'liquid', 'Literal'],
  // EndChangelog
  [/^\{%\s*endchangelog\s*%\}/, 'liquid', 'Literal'],
  // Fake Tabs
  [/^\{%\s*list tabs\s*%\}/, 'liquid', 'Literal'],
  // Fake Tabs end
  [/^\{%\s*endlist\s*%\}/, 'liquid', 'Literal'],
  // ForInLoop
  [/^\{%\s*for\s+[\w.-]+\s+in\s+[\w.-]+\s*%\}/, 'liquid', 'Literal'],
  // EndForInLoop
  [/^\{%\s*endfor\s*%\}/, 'liquid', 'Literal'],
  // Function
  [/^\{\{\s*[\w.-]+?\(.*?\)\s*\}\}/, 'variable', 'Function'],
  // Filter
  [/^\{\{\s*[\w.-]+\s*\|\s*\w+\s*\}\}/, 'variable', 'Filter'],
  // Variable
  [/^\{\{\s*[\w.-]+\s*\}\}/, 'variable', 'Variable'],
  // Attributes
  [
    /^\{\s*(?:[.#](?!T})[a-z0-9_-]+|(?:\s?[a-z0-9_-]+\s*=\s*["']?[a-z0-9_-]+)["']?)+\s*\}/i,
    'liquid',
    'Attributes',
  ],
  // // Space
  // [/^[^\S\r\n]+/, 'text'],
  // Newline
  [/^[\r\n]+/, 'hardbreak'],
  // Text
  // without grabbing liquid/variable syntax
  [/^[\S\s]+?(?={{1,2}|{%)/, 'text'],
  // plain text
  [/^[\S\s]+/, 'text'],
];

export type TokenizerGenerator = Generator<Token | null, void, Token | undefined>;

let variableId = 0;
const vars: Record<string, string> = {};

export class Liquid implements TokenizerGenerator {
  static escape(content: string) {
    return content.replace(/\{\{\s*[\w.-]+\s*\}\}/g, (match) => {
      const id = `${vr}-${variableId++}-v`;
      vars[id] = match;
      return id;
    });
  }

  static unescape(content: string) {
    return content.replace(vrre, (match) => {
      return vars[match];
    });
  }

  private input: string;
  private cursor: number;
  private specification: Specification;

  constructor(input: string, configuration: Configuration = {}) {
    this.input = Liquid.unescape(input);
    this.cursor = 0;

    const {specification = specification_} = configuration;
    this.specification = specification;
  }

  tokenize(this: TokenizerGenerator & this): Token[] {
    this.cursor = 0;

    return Array.from<Token>(this);
  }

  *[Symbol.iterator](this: TokenizerGenerator & this) {
    let value, done;

    do {
      ({value, done} = this.next() ?? {});
      // eslint-disable-next-line no-eq-null, eqeqeq
      if (value == null) {
        return;
      }

      yield value;
    } while (!done);
  }

  next(): IteratorResult<Token | null> {
    const [token, value] = this.match();
    if (!token) {
      return {value: null, done: true};
    }

    if (token.type !== 'text') {
      token.markup = value;
    }
    token.generated = 'liquid';

    this.cursor += value.length;

    return {value: token, done: this.done()};
  }

  return() {
    return this.next();
  }

  throw(): IteratorResult<Token> {
    return {value: null, done: true};
  }

  private match(this: Liquid): [Token, string] | [null] {
    const left = this.input.slice(this.cursor);

    for (const [regexp, type, subtype] of this.specification) {
      const [value] = regexp.exec(left) ?? [null];
      if (value === null) {
        continue;
      }

      switch (type) {
        case 'text':
          return [token(type, {content: value}), value];
        case 'hardbreak':
          return [token(type), value];
        case 'liquid':
          return [token('liquid', {content: '', skip: value, subtype}), value];
        case 'variable':
          return [token('liquid', {content: value, subtype}), value];
        default:
          throw new TypeError('Unexpected liquid token');
      }
    }

    return [null];
  }

  private done() {
    return this.cursor === this.input.length;
  }
}
