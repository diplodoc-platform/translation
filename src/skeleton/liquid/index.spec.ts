/* eslint-disable @typescript-eslint/no-loop-func */
import {Liquid} from '.';
import {condition, filter, functions, loop, text, variable} from './__fixtures__/liquid';

const fixtures = [...text, ...condition, ...functions, ...filter, ...variable, ...loop];

describe('liquid tokenizer smoke', () => {
  it('works', () => {
    const tokenizer = new Liquid('');
    tokenizer.tokenize();
  });
});

describe('liquid tokenizer syntax', () => {
  for (const {section, number, content} of fixtures) {
    test(`${section}-${number}`, () => {
      const tokenizer = new Liquid(content);
      const tokens = [];
      for (const token of tokenizer) {
        tokens.push(token);
      }
      expect(tokens).toMatchSnapshot();
    });
  }
});
