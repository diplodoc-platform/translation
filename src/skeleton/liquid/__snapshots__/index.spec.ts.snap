// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`liquid tokenizer syntax conditions-1 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Person is: ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{% if personType == 'cool' %}",
    "meta": null,
    "nesting": 0,
    "skip": "{% if personType == 'cool' %}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " cool ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{% else %}",
    "meta": null,
    "nesting": 0,
    "skip": "{% else %}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " not cool.",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax conditions-2 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%if var == \\"a\\"%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%if var == \\"a\\"%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " A ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%else%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%else%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " B was the point we examined. Now examine let's examine others.",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax conditions-3 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{% if var == \\"a\\" %}",
    "meta": null,
    "nesting": 0,
    "skip": "{% if var == \\"a\\" %}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "

",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "hardbreak",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Content when var is equal a

",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{% else %}",
    "meta": null,
    "nesting": 0,
    "skip": "{% else %}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "

",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "hardbreak",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Content when var is not equal a

",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{% endif %}",
    "meta": null,
    "nesting": 0,
    "skip": "{% endif %}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "

",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "hardbreak",
  },
]
`;

exports[`liquid tokenizer syntax filter-1 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Hello P",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user.name | capitalize}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user.name | capitalize}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Filter",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "!",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax filter-2 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "We have ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user.name | length}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user.name | length}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Filter",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " users.",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax filter-3 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Users amount: ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{ user.name | length }}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{ user.name | length }}",
    "meta": null,
    "nesting": 0,
    "subtype": "Filter",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": ".",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax filter-4 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Users amount: ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{ user_name-cool | length }}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{ user_name-cool | length }}",
    "meta": null,
    "nesting": 0,
    "subtype": "Filter",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": ".",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax functions-1 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Hello P",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{ user.name.slice(1, 2) }}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{ user.name.slice(1, 2) }}",
    "meta": null,
    "nesting": 0,
    "subtype": "Function",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "vel!",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax functions-2 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Sentence ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user.name.slice(1)}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user.name.slice(1)}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Function",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " content after.",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax functions-3 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user.slice(1)}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user.slice(1)}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Function",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": ".",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax functions-4 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user_name-cool.slice(1)}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user_name-cool.slice(1)}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Function",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": ".",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax loop-1 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{% for x in xs %}",
    "meta": null,
    "nesting": 0,
    "skip": "{% for x in xs %}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " x ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{% endfor %}",
    "meta": null,
    "nesting": 0,
    "skip": "{% endfor %}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
]
`;

exports[`liquid tokenizer syntax loop-2 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%for x in xs%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%for x in xs%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " x ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%endfor%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%endfor%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
]
`;

exports[`liquid tokenizer syntax loop-3 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%for x.a_b in xs.ys%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%for x.a_b in xs.ys%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " x ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%endfor%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%endfor%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
]
`;

exports[`liquid tokenizer syntax loop-4 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%for x.a_b in xs.ys%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%for x.a_b in xs.ys%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "

",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "hardbreak",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{a_b.c}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{a_b.c}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Variable",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "

",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "hardbreak",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{%endfor%}",
    "meta": null,
    "nesting": 0,
    "skip": "{%endfor%}",
    "subtype": "Literal",
    "tag": "",
    "type": "liquid",
  },
]
`;

exports[`liquid tokenizer syntax text-1 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Sentence with plain text. Another plain text sentence.",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax text-2 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Sentence.    Sentence after 4 spaces.",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax variable-1 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Hello ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user.name}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user.name}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Variable",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "!",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax variable-2 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "We have ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{ user.name }}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{ user.name }}",
    "meta": null,
    "nesting": 0,
    "subtype": "Variable",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": " users.",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax variable-3 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Users amount: ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user.name}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user.name}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Variable",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": ".",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;

exports[`liquid tokenizer syntax variable-4 1`] = `
Array [
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "Users amount: ",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": "{{user_name-cool}}",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "{{user_name-cool}}",
    "meta": null,
    "nesting": 0,
    "subtype": "Variable",
    "tag": "",
    "type": "liquid",
  },
  Token {
    "attrs": null,
    "block": false,
    "children": null,
    "content": ".",
    "generated": "liquid",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "text",
  },
]
`;
